### PR TITLE
Add function resource

### PR DESCRIPTION
**Pull request by dump247**
The use case for this resource is setting up datadog agent access. It requires defining a `datadog.explain_statement` function that the dd agent expects to exist. I can setup everything in terraform except for the required function.

https://docs.datadoghq.com/database_monitoring/setup_postgres/aurora/?tab=postgres10

### CODE REVIEW

**Comment by dump247** [comment]
@cyrilgdn Would you be able to take a look at this?

File: README.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ In order to manually run some Acceptance test locally, run the following command
 make testacc_setup 
 
 # Load the needed environment variables for the tests
-source tests/env.sh
+source tests/switch_superuser.sh
```
Comment (dump247) [inline README.md:74]: Unrelated, but I noticed this script name was incorrect in the readme.

File: website/docs/r/postgresql_physical_replication_slot.markdown [hunk 2]
```diff
@@ -19,6 +19,7 @@ any stand-by cluster to replicate physically data.
 resource "postgresql_physical_replication_slot" "my_slot" {
   name  = "my_slot"
 }
+```
```
Comment (dump247) [inline website/docs/r/postgresql_physical_replication_slot.markdown:22]: Unrelated, but I noticed this code block was missing the end marker

**Comment by dump247** [comment]
No worries. Thank you for taking a look. Let me know if there is anything I can do to facilitate the review.

**Comment by dump247** [comment]
@cyrilgdn Any updates?

File: postgresql/resource_postgresql_function.go [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+	}
+
+	sql := fmt.Sprintf("DROP FUNCTION IF EXISTS %s %s", signature, dropMode)
+	if _, err := txn.Exec(sql); err != nil {
+		return err
+	}
+
+	if err = txn.Commit(); err != nil {
+		return fmt.Errorf("Error deleting function: %w", err)
+	}
```
Comment (cyrilgdn) [inline postgresql/resource_postgresql_function.go:214]:
```suggestion
	dropMode := "RESTRICT"
	if v, ok := d.GetOk(funcDropCascadeAttr); ok && v.(bool) {
		dropMode = "CASCADE"
	}

	sql := fmt.Sprintf("DROP FUNCTION IF EXISTS %s %s", signature, dropMode)
	if _, err := db.Exec(sql); err != nil {
                return fmt.Errorf("Error deleting function: %w", err)
	}
```

To simplify the code, if you execute only one SQL query within a transaction, you don't need to create it
(Same comment for all the functions)

**Review by cyrilgdn — COMMENTED** [review]
Just a tiny comment about unnecessary use of transaction that can simplify the code, otherwise it's look good :+1: 
Many thanks for your work on this and sorry again for the response delay.

File: postgresql/resource_postgresql_function.go (hunk 3 above)
Comment (dump247) [reply postgresql/resource_postgresql_function.go]: Done

**Review by cyrilgdn — APPROVED** [review]
@dump247 Thanks a lot for your work on that :+1:

**Comment by cyrilgdn** [comment]
FYI, this has just been released in [v1.16.0](https://registry.terraform.io/providers/cyrilgdn/postgresql/latest)

**Comment by PaleoW** [comment]
Hi @cyrilgdn ,
Thank you for this new postgresql_function ressource !

Would it be possible to add support for the "database" parameter for this ressource as well ? 
Right now, it seems like there is no way to override the provider's database. 

Thank you

**Pull request merged**